File: app/src/main/java/com/orafaaraujo/androiddevconference2017/di/ApplicationModule.java
package com.orafaaraujo.androiddevconference2017.di;

import android.app.Application;
import android.content.Context;

import com.orafaaraujo.androiddevconference2017.DevConferenceApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * {@link Module} exposing the {@link Application} main fields.
 */
@Module
class ApplicationModule {

    private final DevConferenceApplication mApplication;

    ApplicationModule(DevConferenceApplication app) {
        mApplication = app;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return mApplication.getApplicationContext();
    }
}

File: app/src/main/java/com/orafaaraujo/androiddevconference2017/di/ApplicationModule.kt
package com.orafaaraujo.androiddevconference2017.di

import android.app.Application
import android.content.Context
import com.orafaaraujo.androiddevconference2017.DevConferenceApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * [Module] exposing the [Application] main fields.
 */
@Module
class ApplicationModule(val mApplication: DevConferenceApplication) {

    @Provides
    @Singleton
    fun provideApplication() = mApplication

    @Provides
    @Singleton
    fun provideContext(): Context = mApplication.applicationContext
}
